File: from-1501-to-1600/1544-make-the-string-great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        result = []
        for c in s:
            if result and c != result[-1] and c.lower() == result[-1].lower():
                result.pop()
            else:
                result.append(c)

        return ''.join(result)

File: from-1501-to-1600/1544-make-the-string-great/solution.go
func makeGood(s string) string {
    var c string
    result := []string{}
   
    for i := 0; i < len(s); i ++ {
        c = string(s[i])
        if len(result) > 0 && string(result[len(result)-1]) != c && strings.ToLower(c) == strings.ToLower(result[len(result)-1]) {
            result = result[:len(result)-1]
        } else {
            result = append(result, c)
        }
    }
    return strings.Join(result, "")
}